crates/matrix-sdk/src/http_client.rs

```diff
@@ -137,27 +137,22 @@ impl HttpClient {
         config: RequestConfig,
     ) -> Result<http::Request<Bytes>, HttpError> {
         let read_guard;
-        let access_token = if config.force_auth {
-            read_guard = session.read().await;
-            if let Some(session) = read_guard.as_ref() {
-                SendAccessToken::Always(session.access_token.as_str())
-            } else {
-                return Err(HttpError::ForcedAuthenticationWithoutAccessToken);
-            }
-        } else {
-            match Request::METADATA.authentication {
-                AuthScheme::AccessToken => {
-                    read_guard = session.read().await;
+        let access_token = match Request::METADATA.authentication {
```

Comment:
Is this correct? Before the patch we sent an access token regardless of what `METADATA.authentication` told us. 

Now we only set it for the `AccessToken` `AuthScheme`, do the relevant requests set this auth scheme or is  it set to `None` since that's what's in the spec.

Reply:
you're right, I'm testing and this change is not working. I though that it can be done there. I'll update to let it like it was but without returning the error.